src/node/internal/internal_buffer.ts

```diff
@@ -711,25 +753,38 @@ Buffer.prototype.write = function write(string: StringLike,
                                         length?: number | string,
                                         encoding?: string) {
   string = `${string}`;
-  if (arguments.length === 1) {
+  if (offset === undefined) {
     // Buffer#write(string)
     return bufferUtil.write(this, string as string, 0, this.length, "utf8");
   }
 
-  if (typeof offset === 'string') {
-    encoding = offset as string;
-    offset = 0;
+  if (length === undefined && typeof offset === 'string') {
```

Comment:
I know this is how Node does things, but this kind of function overloading emulation sure makes me uneasy.

Reply:
Yep, it's not the best but we're rather stuck with it when implementing these apis